mpl_interactions/controller.py

```diff
@@ -0,0 +1,187 @@
+from IPython.display import display as ipy_display
+from collections import defaultdict
+from .helpers import (
+    create_slider_format_dict,
+    kwarg_to_ipywidget,
+    kwarg_to_mpl_widget,
+    create_mpl_controls_fig,
+    notebook_backend,
+)
+from functools import partial
+from ipywidgets import widgets
+from collections import Iterable
+
+
+class Controls:
+    def __init__(
+        self,
+        slider_formats=None,
+        play_buttons=False,
+        play_button_pos="right",
+        use_ipywidgets=None,
+        out=None,
+        **kwargs
+    ):
+        # it might make sense to also accept kwargs as a straight up arg
+        # to allow for passing the dictionary, but then it would need a different name
+        # and we'd have to combine dicitonarys whihc looks like a hassle
+
+        if out is not None:
+            self.out = out
+        else:
+            self.out = widgets.Output()
+        if use_ipywidgets is None:
+            self.use_ipywidgets = notebook_backend()
+        else:
+            self.use_ipywidgets = use_ipywidgets
+        self.kwargs = kwargs
+        self.slider_format_strings = create_slider_format_dict(slider_formats, self.use_ipywidgets)
+        self.controls = {}
+        self.params = {}
+        self.figs = defaultdict(list)  # maybe should only store weakrefs?
+        self.indices = defaultdict(lambda: 0)
+        self._update_funcs = defaultdict(list)
+        self.controller_list = []
+        self.vbox = widgets.VBox([])
+        self.control_figures = []  # storage for figures made of matplotlib sliders
+        self.add_kwargs(kwargs, slider_formats, play_buttons, play_button_pos)
+
+    def add_kwargs(self, kwargs, slider_formats=None, play_buttons=False, play_button_pos="right"):
+        """
+        If you pass a redundant kwarg it will just be overwritten
+        maybe should only raise a warning rather than an error?
+
+        need to implement matplotlib widgets
+        also a big question is how to dynamically update the display of matplotlib widgets.
```

Comment:
remove this. solution is to just always make a new figure everytime add_kwargs is called